src/Dhall/TypeCheck.hs

```diff
@@ -21,11 +21,13 @@ module Dhall.TypeCheck (
     , TypeMessage(..)
     ) where
 
+import Control.Applicative (liftA2)
 import Control.Exception (Exception)
 import Data.Data (Data(..))
 import Data.Foldable (forM_, toList)
 import Data.Monoid ((<>))
 import Data.Sequence (Seq, ViewL(..))
+import Data.Semigroup (Semigroup(..))
```

Comment:
There is an ambiguous occurrence between the `<>` coming from here and the one above from `Data.Monoid`. I'd recommend removing the one from `Data.Monoid`

Reply:
Also happens in [`src/Dhall.hs`](https://github.com/dhall-lang/dhall-haskell/blob/gabriel/new_map_2/src/Dhall.hs#L93). This seems to be just for GHC-8.2 btw

Reply:
For ghc-7.10.3 too 😉